### ISSUE TITLE
[BUG] Monitor state error

### ISSUE THREAD

**Issue by Kylin-Guo** [apache/hertzbeat#2274]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

1. 某个 ping 连通性监控项目突然同时发生了两封 Email ，一封告警通知，一封恢复通知

![图片](https://github.com/user-attachments/assets/78034a44-35c5-460b-932a-efff60564251)

2. 收到邮件后，登录控制台查看，在可以Ping 通对方主机的情况下，当时监控状态为：宕机

3. 尝试暂停监控后，再恢复监控，监控状态变成正常，但发现在Ping不通且将近30分钟获取不到新数据的情况下，监控状态依然显示正常

![图片](https://github.com/user-attachments/assets/fbf18af7-b60a-4196-801d-bb1ecca4bf1c)

### Environment

```markdown
HertzBeat version(s):v1.6.0
```

**Comment by LiuTianyou** [comment]
Hello, the monitoring status will change only when the trigger times of the threshold rule are reached. For example, if the configured collection period is 10 minutes, the configured trigger times are 3, and the time of the most recent data collection is 10:00:00, and no data has been collected since then, then the monitoring status displayed will be normal at least before 10:30:00.

**Comment by Kylin-Guo** [comment]
> Hello, the monitoring status will change only when the trigger times of the threshold rule are reached. For example, if the configured collection period is 10 minutes, the configured trigger times are 3, and the time of the most recent data collection is 10:00:00, and no data has been collected since then, then the monitoring status displayed will be normal at least before 10:30:00.

Yes, you are right,i thought it will change monitoring status first, and then untill reached trigger times to sent notify
But I dont know why HertzBeat had sent two email in the same time. one alert email, one fix email.

**Comment by Kylin-Guo** [comment]
It doesn't look right.....has been get result for 30 min straight, but the monitoring status didn't switch
![2024-07-18_16-04](https://github.com/user-attachments/assets/994ed67d-9040-4401-86b6-abb53641b65f)
![2024-07-18_16-06](https://github.com/user-attachments/assets/5b375d20-7d79-4cd1-b72d-e7cc09832ed9)
